Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm

```diff
@@ -0,0 +1,201 @@
+{
+    self = [super initWithFrame:frame];
+    if (!self)
+        return nil;
+
+    _webView = view;
+
+    self.alpha = 0;
+    self.userInteractionEnabled = NO;
+    self.layer.allowsGroupOpacity = NO;
+    self.layer.allowsGroupBlending = NO;
+    [self setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+
+    _backdropView = adoptNS([[UIVisualEffectView alloc] initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleLight]]);
+    [_backdropView setFrame:self.bounds];
+    [_backdropView setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+    [[_backdropView layer] setCornerRadius:indicatorCornerRadius];
+    [_backdropView setClipsToBounds:YES];
+    [self addSubview:_backdropView.get()];
+
+    _label = adoptNS([[UILabel alloc] initWithFrame:self.bounds]);
+    [_label setOpaque:NO];
+    [_label setBackgroundColor:nil];
+    [_label setTextAlignment:NSTextAlignmentCenter];
+    [_label setFont:[UIFont boldSystemFontOfSize:indicatorFontSize]];
+    [_label setTextColor:[UIColor blackColor]];
+    [_label setAlpha:indicatorLabelOpacity];
+    [_label setAdjustsFontSizeToFitWidth:YES];
+    [_label setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+    WebCore::PlatformCAFilters::setBlendingFiltersOnLayer([_label layer], WebCore::BlendMode::PlusDarker);
+    [[_backdropView contentView] addSubview:_label.get()];
+
+    [self updatePosition:self.frame];
+    [self setPageCount:pageCount];
+
+    return self;
+}
+
+- (void)dealloc
+{
+    [_timer invalidate];
+    [super dealloc];
+}
+
+- (void)updatePosition:(CGRect)frame
+{
+    RetainPtr webView = _webView.get();
+    if (!webView)
+        return;
+
+    UIEdgeInsets insets = UIEdgeInsetsAdd([webView _computedUnobscuredSafeAreaInset], [webView _computedObscuredInset], UIRectEdgeAll);
+    [self _moveToPoint:UIEdgeInsetsInsetRect(frame, insets).origin animated:NO completionHandler:[view = retainPtr(self)](bool originMoved) {
+        if (!originMoved)
+            return;
+        [view sizeToFit];
+        [view show];
+    }];
+}
+
+- (void)setCurrentPageNumber:(unsigned)currentPageNumber
+{
+    ASSERT(currentPageNumber);
+    bool labelTextShouldChange = _currentPageNumber != currentPageNumber;
+    if (labelTextShouldChange)
+        _currentPageNumber = currentPageNumber;
+    [self _updateLabel:labelTextShouldChange];
+}
+
+- (void)setPageCount:(unsigned)pageCount
+{
+    ASSERT(pageCount);
+    bool labelTextShouldChange = _pageCount != pageCount;
+    if (labelTextShouldChange)
+        _pageCount = pageCount;
+    [self _updateLabel:labelTextShouldChange];
+}
+
+- (void)show
+{
+    [UIView animateWithDuration:indicatorFadeInDuration.seconds() animations:[view = retainPtr(self)] {
+        [view setAlpha:1];
+    }];
+
+    if (_timer)
+        [_timer setFireDate:[NSDate dateWithTimeIntervalSinceNow:indicatorTimeout.seconds()]];
```

Comment:
I wish `WTF::Seconds::operator NSTimeInterval() const { return seconds(); }` was a thing 😞 I'll add it in a future commit